src/Features/LanguageServer/Protocol/LanguageServerTarget.cs

```diff
@@ -445,18 +290,33 @@ public async ValueTask DisposeAsync()
                 disposableLogger.Dispose();
         }
 
-        internal TestAccessor GetTestAccessor()
-            => new TestAccessor(this.Queue);
+        internal TestAccessor GetTestAccessor() => new(this);
 
         internal readonly struct TestAccessor
         {
-            private readonly RequestExecutionQueue _queue;
+            private readonly LanguageServerTarget _server;
+
+            internal TestAccessor(LanguageServerTarget server)
+            {
+                _server = server;
+            }
+
+            internal RequestExecutionQueue.TestAccessor GetQueueAccessor()
+                => _server.Queue.GetTestAccessor();
+
+            internal LspWorkspaceManager.TestAccessor GetManagerAccessor()
+                => _server.Queue.GetTestAccessor().GetLspWorkspaceManager().GetTestAccessor();
+
+            internal RequestDispatcher.TestAccessor GetDispatcherAccessor()
+                => _server.RequestDispatcher.GetTestAccessor();
```

Comment:
i don't love that for testing it seems like we expose all of thsee things.  how come?

Reply:
90% of the tests don't look at this, but there are specific tests that test implementation details of the queue, request dispatcher, or handlers.

For example, a few code action tests look at the handler instance to verify the cached contents match expectations.  And the queue is looked at to verify that the tracked text at each point matches the DidChange requests, or that when we shutdown all the tasks in queue are cancelled.

Since these are TestAccessors the API surface they have access to is still limited, so the tests aren't actually able to arbitrarily access these internal parts.